File: Apps/MobileHarmonica/Domain/Entities/Reed.swift
struct Reed: Equatable {
    let hole: Hole
    let breath: Breath
}

// MARK: - Reed + PositionOnHarmonica

extension Reed {
    init?(at position: PositionOnHarmonica) {
        guard let hole = Hole(at: position) else { return nil }

        self.init(hole: hole, breath: Breath(at: position))
    }
}

File: Web/src/domain/instrument/reed.ts
import type { Breath } from './breath.js'
import type { Hole } from './hole.js'

export interface Reed {
    readonly hole: Hole
    readonly breath: Breath
}

export function isSameReed(one: Reed, other: Reed): boolean {
    return one.hole === other.hole && one.breath === other.breath
}
